delta/plugins/storage/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/storage/storages/operations/s3/S3LocationGenerator.scala

```diff
@@ -0,0 +1,17 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.storage.storages.operations.s3
+
+import akka.http.scaladsl.model.Uri
+import ch.epfl.bluebrain.nexus.delta.rdf.syntax.uriSyntax
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.ProjectRef
+
+import java.util.UUID
+
+class S3LocationGenerator(prefix: Uri) {
```

Comment:
So it is used for upload and the batch and also later for the delegate registration where we will have to generate a path